### PR TITLE
add delete comment functionality

### PR DESCRIPTION
**Pull request by menynexi**
I added a DeleteCommentServlet that handles deleted comments when given an id, I also re-wrote the comment-script file to pass only a comment. Additionally I separated the addition of text and the addition of a delete button with separate helper functions. 

https://mangutierrez-step-20202.uc.r.appspot.com

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for deleting tasks. */
+@WebServlet("/delete-comment")
+public class DeleteCommentServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long id = Long.parseLong(request.getParameter("id"));
+
```
Comment (lattai) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:34]: Can get rid of extra line

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    Key taskEntityKey = KeyFactory.createKey("Comment", id);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.delete(taskEntityKey);
+  }
+}
```
Comment (lattai) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:39]: New line at End of file

File: portfolio/src/main/webapp/commentScript.js [hunk 3]
```diff
@@ -33,13 +33,35 @@ function addCommentsToDom(commentList){
...
-            createComment(commentList[i].username, commentList[i].reflection, commentList[i].timeStamp));
+            createComment(commentList[i]));
     }
 }
 
-/** Creates an <li> element containing author: comment. */
-function createComment(user, reflection, timeStamp) {
+/** Creates an <li> element containing comments. */
+function createComment(comment) {
   const liElement = document.createElement('li');
```
Comment (lattai) [inline portfolio/src/main/webapp/commentScript.js:42]: Make 'li' private string variable

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for deleting tasks. */
+@WebServlet("/delete-comment")
+public class DeleteCommentServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long id = Long.parseLong(request.getParameter("id"));
```
Comment (lzoebaker) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:33]: Could replace string literals like "id" and "Comment" with private final variables

File: portfolio/src/main/webapp/commentScript.js (hunk 3 above)
Comment (lattai) [reply portfolio/src/main/webapp/commentScript.js:42]: and 'Comment", "Delete", "id". Maybe "\n"'s

File: portfolio/src/main/webapp/commentScript.js [hunk 5]
```diff
@@ -33,13 +33,35 @@ function addCommentsToDom(commentList){
     commentContainer.innerHTML = '';
     for (let i = 0; i < commentList.length; i++){
```
Comment (lzoebaker) [inline portfolio/src/main/webapp/commentScript.js:34]: Space before the {

File: portfolio/src/main/webapp/commentScript.js [hunk 6]
```diff
@@ -33,13 +33,35 @@ function addCommentsToDom(commentList){
...
-  liElement.innerText = user + "--> " + reflection + "\n" + timeStamp + "\n\n\n";
+  liElement.className = 'Comment';
+
+  const commentElement = document.createElement('span');
+  commentElement.innerText = comment.username + "--> " + comment.reflection + "\n" + comment.timeStamp + "\n\n\n";
+  
+  const deleteButtonElement = document.createElement('button');
+  deleteButtonElement.innerText = 'Delete';
+  deleteButtonElement.addEventListener('click', () => {
+    deleteComment(comment);
```
Comment (lzoebaker) [inline portfolio/src/main/webapp/commentScript.js:51]: Could you maybe move the creation of the deleteButton into a separate method?

File: portfolio/src/main/webapp/commentScript.js [hunk 7]
```diff
@@ -29,17 +43,48 @@ function handleResponse(response){
...
+            createComment(commentList[i]));
     }
 }
 
-/** Creates an <li> element containing author: comment. */
-function createComment(user, reflection, timeStamp) {
-  const liElement = document.createElement('li');
-  liElement.innerText = user + "--> " + reflection + "\n" + timeStamp + "\n\n\n";
+/** Creates an <li> element containing comments. */
+function createComment(comment) {
```
Comment (DirtGrubDylan) [inline portfolio/src/main/webapp/commentScript.js:55]: At this point, I think this method name should be something like `commentsContainer`

File: portfolio/src/main/java/com/google/sps/servlets/Comment.java [hunk 8]
```diff
@@ -29,4 +30,12 @@ public long getId() {
   public String getTimeStamp() {
     return this.timeStamp;
   }
+
+  public void setMaxComment(int maxComment) {
```
Comment (WillHipschman) [inline portfolio/src/main/java/com/google/sps/servlets/Comment.java:34]: What uses these? It seems weird that a comment object has a method that returns the maximum number of comments...

Comment (menynexi) [reply portfolio/src/main/java/com/google/sps/servlets/Comment.java:34]: Nothing uses that it was an idea to set a maxComment value to a comment that could later be used in the commentScript to display comments. However, it turns out that the problem is non existent in the Dev App Server and only exist on the public server so I will talk to support about it and if it cant be fix I plan to use this implementation

**Pull request merged**